### DISCUSSION TITLE
Workflow for handling issues, pull requests and main branch

### DISCUSSION THREAD

**Discussion by nunoguedelha** [ami-iit/openmct#3] [General]
We just forked this project from the `nasa` organization, which is one of the main dependencies of the https://github.com/ami-iit/yarp-openmct project. The motivation for having such fork is to:
- submit issues that specifically affect the `yarp-openmct` project in the context of its use within IIT AMI lab,
- drive derived internal discussions, before reporting the problem to the original repository,
- implement and deploy fixes or required new features faster, before they are approved by the original repository maintainers,

For that purpose, we shall agree on a workflow for handling issues and pull requests, as suggested below:
- For every encountered problem or bug believed to be originating in this repository, a new issue shall be opened here first.
- After an initial assessment by the maintainers (@nunoguedelha , @traversaro ), a duplicate of the issue (digested as required) shall be opened in https://github.com/nasa/openmct.
- It might happen that some trivial issue doesn't require any specific assessment and can be opened directly on https://github.com/nasa/openmct. In such case, a duplicate can be opened here as a proxy link to the original issue.
- In any case, if necessary, a respective fix/feature PR can be submitted here, on a branch local to this repository.
- If the fix/feature is deemed to be acceptable to the original repository (https://github.com/nasa/openmct), so typically not a workaround, a duplicate PR shall be submitted to https://github.com/nasa/openmct through a different dedicated branch.

@traversaro , I hope the statements listed above reflect well our initial discussion on the issues/PRs workflow. Feel free to comment and propose changes along the way, and invite anyone else of interest to this discussion.

**Comment by nunoguedelha** [comment]
CC @riccardogrieco, @S-Dafarra

**Comment by traversaro** [comment]
> @traversaro , I hope the statements listed above reflect well our initial discussion on the issues/PRs workflow.

Yes, I think this is a good recap, thanks!

**Comment by nunoguedelha** [comment]
CC @riccardogrieco @S-Dafarra
